gammapy/estimators/ts_map.py

```diff
@@ -31,6 +34,32 @@ def round_up_to_odd(f):
     return int(np.ceil(f) // 2 * 2 + 1)
 
 
+def compute_reco_exposure(dataset, spectral_model=None):
+    """
+    Create and exposure map in reco energies
+    Parameters
+    ----------
+    dataset:`~gammapy.cube.MapDataset` or `~gammapy.cube.MapDatasetOnOff`
+            the input dataset
+    spectral_model: `~gammapy.modeling.models.SpectralModel`
+            assumed spectral shape. If none, a Power Law of index 2 is assumed
+    """
+    if spectral_model is None:
+        spectral_model = PowerLawSpectralModel()
+    model = SkyModel(
+        spatial_model=ConstantFluxSpatialModel(), spectral_model=spectral_model
+    )
+    kernel = None
+    if dataset.edisp is not None:
+        kernel = dataset.edisp.get_edisp_kernel(position=dataset._geom.center_skydir)
+    meval = MapEvaluator(model=model, exposure=dataset.exposure, edisp=kernel)
+    npred = meval.compute_npred()
+    e_reco = dataset._geom.get_axis_by_name("energy").edges
+    ref_flux = spectral_model.integral(e_reco[:-1], e_reco[1:])
```

Comment:
There is one thing I don't understand yet: in the old `_map_spectrum_weight` we divided by the total flux in true energy, here in reco energy. I wonder what the correct thing to do is here, because the energy ranges are not the necessarily the same? Compute the flux in true energy, apply the edisp and divide? (maybe @registerrier can help here...)

Reply:
I would say this looks correct to me.

If you measure `n_excess` and if your model with `amplitude_0` normalization predicts `npred` counts, your actual flux (assuming the model is correct) is `amplitude_0 * n_excess/npred`.
This is what is done here.

The exposure in reco energy is simply the ratio between the assumed flux and its associated `npred` in the considered reco energy range.

Reply:
Thanks, now I see. So it's basically equivalent to what we do for flux point estimation, which is good. The norm with respect to the reference model is fitted in a defined reco energy range and the flux is computed as `norm * flux_ref`, where `flux_ref` is computed again using the repo energy range. Makes sense.

I already started to refactor the `TSMapEstimator` to use a `norm` factor internally probably we should develop this further to make it fully equivalent to what we do for flux point estimation...